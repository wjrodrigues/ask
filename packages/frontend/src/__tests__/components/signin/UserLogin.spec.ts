import { describe, it, expect, vi } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";

import UserLoginVue from "@/components/signin/UserLogin.vue";
import { base_urls, basic_mount, faker, nock } from "@/__tests__/help";

describe("when render UserLoginVue", () => {
  it("render form with all components", () => {
    const wrapper = shallowMount(
      UserLoginVue,
      basic_mount({ props: {}, plugins: [] })
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it("validate required input", async () => {
    const wrapper = mount(
      UserLoginVue,
      basic_mount({ props: {}, plugins: [] })
    );

    await wrapper.findAll("input").forEach(async (e) => e.setValue(""));
    await wrapper.trigger("change");

    const error_messages = await wrapper.findAll(".v-messages > div");

    expect(error_messages.length).toEqual(2);
    error_messages.map((msg) =>
      expect(msg.text()).toEqual("Campo é obrigatório")
    );
  });

  it("submit form button", async () => {
    const wrapper = mount(
      UserLoginVue,
      basic_mount({ props: {}, plugins: [] })
    );

    wrapper.find("[name='email']").setValue(faker.internet.email());
    wrapper.find("[name='password']").setValue(faker.internet.password());

    await wrapper.trigger("change");

    expect(wrapper.vm.errors).toEqual({
      email: "",
      password: "",
    });
    expect(wrapper.vm.form).toEqual(true);
    await wrapper.find("form").trigger("submit");
  });

  it("go to sign up", async () => {
    const spyPush = { push: () => {} };
    const spy = vi.spyOn(spyPush, "push");

    const wrapper = mount(
      UserLoginVue,
      basic_mount({
        props: {},
        plugins: [],
        mocks: {
          $router: {
            push: spy,
          },
        },
      })
    );

    const signin = wrapper.findAll("button")[0];
    expect(signin.text()).toEqual("Inscrever-se");

    await signin.trigger("click");
    await wrapper.trigger("change");

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith("/signup");
  });
});
