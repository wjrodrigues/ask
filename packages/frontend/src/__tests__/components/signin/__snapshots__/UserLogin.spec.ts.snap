// Vitest Snapshot v1

exports[`when render UserLoginVue > render form with all components 1`] = `
<v-card-stub
  border="false"
  class="mx-auto"
  density="default"
  disabled="false"
  exact="false"
  flat="false"
  hover="false"
  loading="false"
  replace="false"
  ripple="false"
  tag="div"
  title="Entrar"
  variant="elevated"
  width="344"
/>
`;
